cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java

```diff
@@ -336,19 +351,39 @@ public void execute() {
 
-        TaskExecutor taskExecutor = new TaskExecutor.TaskBuilder()
-                .addTask(new CleanTargetCacheDirTask(), isSingleFile) // clean the target cache dir(projects only)
-                .addTask(new ResolveMavenDependenciesTask(outStream)) // resolve maven dependencies in Ballerina.toml
+        CleanTargetCacheDirTask cleanTargetCacheDirTask = new CleanTargetCacheDirTask();
+        ResolveMavenDependenciesTask resolveMavenDependenciesTask = new ResolveMavenDependenciesTask(outStream);
+        CompileTask compileTask = new CompileTask(outStream, errStream, false, isPackageModified,
+                buildOptions.enableCache());
+        RunTestsTask runTestsTask = new RunTestsTask(outStream, errStream, rerunTests, groupList, disableGroupList,
+                testList, includes, coverageFormat, moduleMap, listGroups, excludes, cliArgs);
+
+        CreateTestExecutableTask createTestExecutableTask = null;
+
+        if (!project.buildOptions().cloud().isEmpty()) {
+            //if cloud flag is set, create the test executable
+            createTestExecutableTask = new CreateTestExecutableTask(outStream, this.output, runTestsTask);
+        }
+
+        RunNativeImageTestTask runNativeImageTestTask = new RunNativeImageTestTask(outStream, rerunTests, groupList,
+                disableGroupList, testList, includes, coverageFormat, moduleMap, listGroups);
+        DumpBuildTimeTask dumpBuildTimeTask = new DumpBuildTimeTask(outStream);
+
+        TaskExecutor.TaskBuilder taskBuilder = new TaskExecutor.TaskBuilder();
+
+        taskBuilder.addTask(cleanTargetCacheDirTask, isSingleFile) // clean the target cache dir(projects only)
+                .addTask(resolveMavenDependenciesTask) // resolve maven dependencies in Ballerina.toml
                 // compile the modules
-                .addTask(new CompileTask(outStream, errStream, false, isPackageModified, buildOptions.enableCache()))
-//                .addTask(new CopyResourcesTask(), listGroups) // merged with CreateJarTask
-                .addTask(new RunTestsTask(outStream, errStream, rerunTests, groupList, disableGroupList, testList,
-                        includes, coverageFormat, moduleMap, listGroups, excludes, cliArgs),
-                        project.buildOptions().nativeImage())
-                .addTask(new RunNativeImageTestTask(outStream, rerunTests, groupList, disableGroupList,
-                        testList, includes, coverageFormat, moduleMap, listGroups),
-                        !project.buildOptions().nativeImage())
-                .addTask(new DumpBuildTimeTask(outStream), !project.buildOptions().dumpBuildTime())
+                .addTask(compileTask);
+
+        if (!project.buildOptions().cloud().isEmpty()
+                && createTestExecutableTask != null) {
+            taskBuilder.addTask(createTestExecutableTask);
+        }
+        //                .addTask(new CopyResourcesTask(), listGroups) // merged with CreateJarTask
```

Comment:
Do we need this comment?

Reply:
This was there before the modification. While the comment does not match with the current implementation (the code structure), this task might have some relevance?